site/public/js/userform.js

```diff
@@ -282,11 +305,12 @@ function clearValidityWarnings() {
 }
 
 function closeButton() {
-    $('#edit-user-form').css('display', 'none');
-    if($('[name="edit_user"]').val() == 'true') {
+    // If form contain some data, prompt the user for data loss
+    if(!isUserFormEmpty() && confirm('Changes to the Form will be lost!')){
         $('[name="user_id"]', $("#edit-user-form")).val("");
         clearUserFormInformation();
     }
+    $('#edit-user-form').css('display', 'none');
```

Comment:
I think we want that this should only happen if we confirm. Right now, if we hit cancel after we get the confirm box, the modal will still close, but the data will remain the next time we open the modal.

Reply:
Added changes and rebased.
PTAL, Thanks!